src/base.cpp

```diff
@@ -47,12 +47,13 @@
 
 namespace lbann {
 
-lbann_comm* initialize(int& argc, char**& argv, int seed) {
+world_comm_ptr initialize(int& argc, char**& argv, int seed) {
   // Initialize Elemental.
   El::Initialize(argc, argv);
   // Create a new comm object.
   // Initial creation with every process in one model.
-  auto* comm = new lbann_comm(0);
+  auto comm = world_comm_ptr{new lbann_comm(0), &lbann::finalize };
```

Comment:
Should we be doing this with `make_unique` or something?

Reply:
You cannot use `make_unique` with a custom deleter. The return type of `make_unique<T>` is explicitly `unique_ptr<T,std::default_delete<T>>`. In lieu of a custom `make_unique` implementation that accommodates deleters in some fashion, this is the best that can be done.